### PR TITLE
Check signature weight when loading block or header by height.

### LINKED ISSUE #1655: Never Return A Block/BlockHeader With Insufficient Finality Signatures To a Fetcher
When a participating node receives a fetcher request for a `Block` or `BlockHeader` by height, it should never return a block header with insufficient finality signature weight.

If it does, it should be banned.

### PR DESCRIPTION
**Pull request by afck**
Closes #1655.

### CODE REVIEW

File: node/src/components/linear_chain_sync/operations.rs [hunk 1]
```diff
@@ -193,7 +193,18 @@ fn validate_finality_signatures(
...
     // Cryptographically verify block signatures
     block_signatures.verify()?;
+    weigh_finality_signatures(
+        trusted_validator_weights,
+        finality_threshold_fraction,
+        block_signatures,
+    )
+}
 
+pub(crate) fn weigh_finality_signatures(
```
Comment (goral09) [inline node/src/components/linear_chain_sync/operations.rs:203]: Can you add an API doc comment to this method?

File: node/src/reactor/validator.rs [hunk 2]
```diff
@@ -783,36 +782,51 @@ impl reactor::Reactor for Reactor {
...
+                                match Message::new_get_response(
+                                    &fetch_or_not_found_header_with_sigs,
+                                ) {
+                                    Ok(message) => {
+                                        effect_builder.send_message(sender, message).await
+                                    }
+                                    Err(error) => error!("failed to create get-response {}", error),
                                 }
-                            };
+                            }
```
Comment (goral09) [inline node/src/reactor/validator.rs:828]: This looks very similar to the method in `linear_chain_sync::operations`, is that what the `TODO` there is about?

Comment (afck) [reply node/src/reactor/validator.rs:828]: It looks like I'll probably have to rewrite this PR anyway after `feat-fast-sync` has been rebased and @xcthulhu's latest refactorings merged. Then I'll try to deduplicate this, too.

**Comment by afck** [comment]
Closing this since we're rewriting it. We'll take your comments into account!

**Pull request closed without merging**